analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/SonarDiagnosticAnalyzer.cs

```diff
@@ -27,6 +27,7 @@ public abstract class SonarDiagnosticAnalyzer : DiagnosticAnalyzer
         public static readonly string EnableConcurrentExecutionVariable = "SONAR_DOTNET_ENABLE_CONCURRENT_EXECUTION";
 
         protected virtual bool EnableConcurrentExecution => IsConcurrentExecutionEnabled();
+        protected virtual Func<Compilation, bool> ShouldRegisterAnalyzerPredicate => null;
```

Comment:
Lambda is painful to use, and the double handling is kindof hacky.

What about new abstract class with abstract member to override, to make the contract more clear?